### ISSUE TITLE
Can't overide default ignore patterns when using --stdin-filename containing a .git folder

### ISSUE THREAD

**Issue by tfaulkner-eaglegroup** [eslint/eslint#10059]
I can't figure out how to override default ignore patterns when using the `--stdin-filename` flag. No matter what I do, eslint seems determined to ignore hidden files when processed by stdin.

It should be pretty easy to reproduce. Drop a file in to .git and try something like `cat .git/hello.js | eslint --stdin --stdin-filename .git/hello.js --ignore-pattern '!.git/hello.js`. I also tried `--no-ignore`.

I *am* on Windows at the moment, so I'm not sure if that's a factor...

My use case is that I want to lint my staged commits using vim + fugitive. ALE passes the buffer over stdin with a `--stdin-filename` flag of `fugitive://path/to/project/.git//0/path/to/file.js`. Looks like eslint is choking on the `.git` part of the path. I can work around it by renaming the buffer, but I thought it was interesting enough that I'd open an issue. 

---

**Tell us about your environment**

* **ESLint Version:** v4.18.2
* **Node Version:** v.8.9.4
* **npm Version:** 5.3.0
* **OS:** Windows 10 (not sure if related -- haven't tested this problem on Linux)

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:** 

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "extends": "standard"
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

**Code:**

```js
console.log("hello world!");
```
**Command:**

```powershell
Get-Content .git\hello.js | eslint --stdin --stdin-filename .git\hello.js --ignore-pattern '!.git\hello.js'
```

**What did you expect to happen?**
I expected eslint to lint the file
**What actually happened? Please include the actual, raw output from ESLint.**

```bash
  0:0  warning  File ignored by default.  Use a negated ignore pattern (like "--ignore-pattern '!<relative/path/to/filename>'") to over
ride

✖ 1 problem (0 errors, 1 warning)
```

**Comment by tfaulkner-eaglegroup** [comment]
Hmm, I can't reproduce my example this morning. Doing some more investigating.

**Comment by tfaulkner-eaglegroup** [comment]
Ok, looks like .git has to be a directory in the `--stdin-filename` parameter for this to show up. No idea if I was just losing my sanity last night or what, but I've updated my issue.

**Comment by tfaulkner-eaglegroup** [comment]
The .git folder itself is ignored, and [it is not possible to re-include a file if a parent directory of that file is excluded](https://github.com/kaelzhang/node-ignore/blob/9f4e1e9b34a7c3070c6e0082a0e32fb9645f5c85/index.js#L141).

`--ignore-pattern '!.git'` works.
